components/device-sync.js

```diff
@@ -0,0 +1,269 @@
+  const [debugValue, setDebugValue, clearValue] = useVault(me, 'debug')
+
+  const manage = useCallback(async () => {
+    if (enabled && connected) {
+      showModal((onClose) => (
+        <div>
+          <h2>Device sync is enabled!</h2>
+          <p>
+            Sensitive data (like wallet credentials) is now securely synced between all connected devices.
+          </p>
+          <p className='text-muted text-sm'>
+            Disconnect to prevent this device from syncing data or to reset your passphrase.
+          </p>
+          <div className='d-flex justify-content-between'>
+            <div className='d-flex align-items-center ms-auto gap-2'>
+              <Button className='me-2 text-muted nav-link fw-bold' variant='link' onClick={onClose}>close</Button>
+              <Button
+                variant='primary'
+                onClick={() => {
+                  disconnectVault()
+                  onClose()
+                }}
+              >disconnect
+              </Button>
+            </div>
+          </div>
+        </div>
+      ))
+    } else {
+      showModal((onClose) => (
+        <ConnectForm onClose={onClose} onConnect={onConnect} enabled={enabled} />
+      ))
+    }
+  }, [migrate, enabled, connected, value])
+
+  const reset = useCallback(async () => {
+    const schema = yup.object().shape({
+      confirm: yup.string()
+        .oneOf(['yes'], 'you must confirm by typing "yes"')
+        .required('required')
+    })
+    showModal((onClose) => (
+      <div>
+        <h2>Reset device sync</h2>
+        <p>
+          This will delete all encrypted data on the server and disconnect all devices.
+        </p>
+        <p>
+          You will need to enter a new passphrase on this and all other devices to sync data again.
+        </p>
+        <Form
+          className='mt-3'
+          initial={{ confirm: '' }}
+          schema={schema}
+          onSubmit={async values => {
+            await clearVault()
+            onClose()
+          }}
+        >
+          <Input
+            label='This action cannot be undone. Type `yes` to confirm.'
+            name='confirm'
+            placeholder=''
+            required
+            autoFocus
+            autoComplete='off'
+          />
+          <div className='d-flex justify-content-between'>
+            <div className='d-flex align-items-center ms-auto'>
+              <CancelButton onClick={onClose} />
+              <SubmitButton variant='danger'>
+                continue
+              </SubmitButton>
+            </div>
+          </div>
+        </Form>
+      </div>
+    ))
+  }, [])
+
+  const onConnect = useCallback(async (values, formik) => {
+    if (values.passphrase) {
+      try {
+        await setVaultKey(values.passphrase)
+        await migrate()
+        apollo.cache.evict({ fieldName: 'BestWallets' })
+        apollo.cache.gc()
+        await apollo.refetchQueries({ include: ['BestWallets'] })
```

Comment:
I'm just browsing right now and haven't taken a close look, but I suspect we'll eventually want the vault to not be this tightly "coupled" with wallets.